### PR TITLE
Add delegator public

### PR DESCRIPTION
**Pull request by sukantoraymond**

### CODE REVIEW

File: cmd/subnetcmd/join.go [hunk 1]
```diff
@@ -444,32 +444,43 @@ func checkIsValidating(subnetID ids.ID, nodeID ids.NodeID, pClient platformvm.Cl
...
 		// Get NodeIDs of all validators on the subnet
 		validators, err := subnet.GetSubnetValidators(subnetID)
 		if err != nil {
 			return ids.EmptyNodeID, err
 		}
 		// construct list of validators to choose from
 		var validatorList []string
-		fmt.Printf("defaultLocalNetworkNodeIDs %s \n", defaultLocalNetworkNodeIDs)
-
 		for _, localNodeID := range defaultLocalNetworkNodeIDs {
```
Comment (felipemadero) [inline cmd/subnetcmd/join.go:464]: should do something else for fuji?

File: cmd/subnetcmd/join.go [hunk 2]
```diff
@@ -444,32 +444,43 @@ func checkIsValidating(subnetID ids.ID, nodeID ids.NodeID, pClient platformvm.Cl
...
+						nodeIDFound = true
+						break
+					}
 				}
 			}
-			if !nodeIDFound {
-				fmt.Printf("adding validators %s \n", localNodeID)
 
+			if !nodeIDFound {
 				validatorList = append(validatorList, localNodeID)
```
Comment (felipemadero) [inline cmd/subnetcmd/join.go:476]: should only add validators if delegating?

File: cmd/subnetcmd/join.go [hunk 3]
```diff
@@ -481,55 +492,71 @@ func promptNodeIDToAdd(subnetID ids.ID) (ids.NodeID, error) {
...
+					},
+				},
+			)
+		}
 	}
-	ctx := context.Background()
-	pClient := platformvm.NewClient(constants.LocalAPIEndpoint)
-	walletBalance, err := getAssetBalance(ctx, pClient, ewoqPChainAddr, esc.AssetID)
+	ux.Logger.PrintToUser("What amount of the subnet native token would you like to stake?")
+	initialSupply, err := app.Prompt.CaptureUint64("Stake amount")
```
Comment (felipemadero) [inline cmd/subnetcmd/join.go:555]: should do a similar check for fuji than for local

File: pkg/subnet/public.go [hunk 4]
```diff
@@ -530,6 +554,43 @@ func (d *PublicDeployer) createTransformSubnetTX(
...
+	recipientAddr ids.ShortID,
+	stakeAmount uint64,
+	subnetID ids.ID,
+	nodeID ids.NodeID,
+	assetID ids.ID,
+	startTime uint64,
+	endTime uint64,
+	wallet primary.Wallet,
+) (ids.ID, error) {
+	options := d.getMultisigTxOptions([]ids.ShortID{})
```
Comment (felipemadero) [inline pkg/subnet/public.go:567]: I already commented on this on another PR

File: cmd/subnetcmd/addDelegator.go [hunk 5]
```diff
@@ -0,0 +1,224 @@
+// Copyright (C) 2022, Ava Labs, Inc. All rights reserved.
```
Comment (felipemadero) [inline cmd/subnetcmd/addDelegator.go:1]: would prefer two different PRs to have no so much code in common

**Review by felipemadero — COMMENTED** [review]
can this PR have less code in common with the other one? so as to check the changes

**Review by dasconnor — APPROVED** [review]

**Pull request merged**